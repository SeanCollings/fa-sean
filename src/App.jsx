import React from 'react';
import styled from 'styled-components';
import { BrowserRouter, Redirect, Route, Switch } from 'react-router-dom';
import { Provider as ReduxProvider } from 'react-redux';
import Footer from './components/footer';
import Header from './components/header';
import HomePage from './components/pages/home-page';
import { store } from './state';

const SAppContainer = styled.div`
  position: relative;
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  min-height: 100vh;
  font-family: 'Proxima Nova', 'Helvetica Neue', Helvetica, Helvetica, Arial,
    sans-serif;
`;
const SBodyContainer = styled.div`
  position: relative;
  margin-top: 4rem;
  padding: 1.5rem 1.5rem 0;
`;

const App = () => {
  return (
    <ReduxProvider store={store}>
      <SAppContainer>
        <BrowserRouter>
          <Header />
          <SBodyContainer>
            <Switch>
              <Route exact path="/" component={HomePage} />
              <Route path="*" render={() => <Redirect to="/" />} />
            </Switch>
          </SBodyContainer>
          <Footer />
        </BrowserRouter>
      </SAppContainer>
    </ReduxProvider>
  );
};

export default App;
