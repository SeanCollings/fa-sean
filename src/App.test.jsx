import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('App unit test', () => {
  test('should render', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
