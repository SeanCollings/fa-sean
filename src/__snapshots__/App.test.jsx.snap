// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App unit test should render 1`] = `
<div
  className="sc-fKFyDc jZUdTc"
>
  <div
    className="sc-gKsewC ezMuak"
  >
    <div
      className="sc-pFZIQ dKPBra"
    >
      <div
        className="sc-iBPRYJ igIkBL"
      >
        <div
          className="sc-hKgILt cvTavO"
        >
          <div
            className="sc-eCssSg cGdqlL"
            onClick={[Function]}
          >
            <div>
              Women
            </div>
            <div
              className="sc-jSgupP hDMCDB"
            />
          </div>
          <div
            className="sc-eCssSg cGdqlL"
            onClick={[Function]}
          >
            <div>
              Men
            </div>
          </div>
          <div
            className="sc-eCssSg cGdqlL"
            onClick={[Function]}
          >
            <div>
              Kids
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-jrAGrp czFpDo"
      >
        <div
          className="sc-fubCfw gfiPIt"
          onClick={[Function]}
        >
          Fa-sean
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-bBXqnf iyKlty"
  >
    <div>
      <div
        className="sc-dIUggk eHYCqR"
      >
        Women
        's Clothing
      </div>
      <div
        className="sc-crrsfI eOJQly"
      >
        <div
          className="sc-dQppl exHpdh"
        >
          <select
            className="sc-iqHYGH fdGNPn"
            id="sort"
            name="sort"
            onChange={[Function]}
          >
            <option
              value="popularity"
            >
              Popularity
            </option>
            <option
              value="new"
            >
              New Arrivals
            </option>
            <option
              value="price_high"
            >
              Price High to Low
            </option>
            <option
              value="price_low"
            >
              Price Low to High
            </option>
            <option
              value="brand_asc"
            >
              Brand A to Z
            </option>
            <option
              value="brand_desc"
            >
              Brand Z to A
            </option>
          </select>
        </div>
      </div>
      <div
        className="sc-kfzAmx aYMdN"
      />
      <div>
        No results
      </div>
    </div>
  </div>
  <footer
    className="sc-bdfBwQ fvPrKx"
  >
    <div
      className="sc-gsTCUz bFsUWP"
    >
      <div>
        <a
          className="sc-dlfnbm cDFxQC"
          href="https://www.theiconic.com.au/"
          target="_blank"
        >
          The Iconic
        </a>
      </div>
    </div>
  </footer>
</div>
`;
